### PR TITLE
Update combat.py

### PR DESCRIPTION
**Pull request by tjuHaoXiaotian**
Fix some bugs:
```
    1. In get_agent_obs(), the local variable '_type' was mistakenly written as 'type'.
    2. In get_agent_obs(), the x-coordinates and y-coordinates of the entites (whether the entity is 'agent' or 'opponent') within each agent's view range were incorrectly set to the agent's position.
    3. self.agent_prev_pos and self.opp_prev_pos are not updated, but in __update_agent_view() and __update_opp_view(), these previous positions are cleared. Thus the movement records are incorrect.
    4. The self._agent_cool and self._opp_cool are not updated (i.e., the agents and opponents could execute 'attack' action at every step if within range).
        updates:
          (1) update the is_fireable() function to check if the attack agents have been cooled down. 
          (2) add self._agent_cool_step and self._opp_cool_step to track the cooling down of the agents and opponents.
    5. In the opps_action() function, the dead opponent could still call reduce_distance_move() function to sample a 'move' action, which will raise an 'no where to move' exception when the position of the dead opponent is the same with any of the agents.
    6. Update the get_agent_obs(): only return the local view of observation when the agent is alive, otherwise return a zero vector.
```

Add 2 new functions to support CTDE
```
    7. Add get_state() to support CTDE (centralized training with decentralized execution).
```

### CODE REVIEW

File: ma_gym/envs/combat/combat.py [hunk 1]
```diff
@@ -45,38 +45,47 @@ class Combat(gym.Env):
...
     def __init__(self, grid_shape=(15, 15), n_agents=5, n_opponents=5, init_health=3, full_observable=False,
-                 step_cost=0, max_steps=100):
+                 step_cost=0, max_steps=100, step_cool=1, render=True):
         self._grid_shape = grid_shape
         self.n_agents = n_agents
         self._n_opponents = n_opponents
         self._max_steps = max_steps
+        self._step_cool = step_cool + 1
         self._step_cost = step_cost
+        self._render = render
```
Comment (koulanurag) [inline ma_gym/envs/combat/combat.py:55]: I don't see it being used anywhere.

**Review by koulanurag — COMMENTED** [review]
Overall looks good! Thanks for the changes.

I can merge it once you resolve the "_render" comment.

File: ma_gym/envs/combat/combat.py (hunk 1 above)
Comment (tjuHaoXiaotian) [reply ma_gym/envs/combat/combat.py]: Yes, the "_render" is only for personal usage. I have removed this unused attribute.

**Comment by koulanurag** [comment]
All looks good. But, Let's write some more exhaustive test cases. You may use [this](https://github.com/koulanurag/ma-gym/blob/master/tests/envs/test_checkers.py) as a reference.

**Pull request merged**